store/cachemulti/store.go

```diff
@@ -173,13 +153,88 @@ func (cms Store) Write() {
 	}
 }
 
-// Unlock calls Unlock on each underlying LockingStore.
-func (cms Store) Unlock() {
-	for _, store := range cms.stores {
-		if s, ok := store.(types.LockingStore); ok {
-			s.Unlock()
+// Lock, Unlock, RLockRW, LockRW, RUnlockRW, UnlockRW constitute a permissive locking interface
+// that can be used to synchronize concurrent access to the store. Locking of a key should
+// represent locking of some part of the store. Note that improper access is not enforced, and it is
+// the user's responsibility to ensure proper locking of any access by concurrent goroutines.
+//
+// Common mistakes may include:
+//   - Introducing data races by reading or writing state that is claimed by a competing goroutine
+//   - Introducing deadlocks by locking in different orders through multiple calls to locking methods.
+//     i.e. if A calls Lock(a) followed by Lock(b), and B calls Lock(b) followed by Lock(a)
+//   - Using a key as an exclusive lock after it has already been initialized as a read-write lock
```

Comment:
Question, what's the expected behavior here? Casting the `sync.RWMutex` to a `sync.Mutex` should fail right?

Reply:
yeah failed type assertion will panic